sys/net/application_layer/dhcpv6/client.c

```diff
@@ -127,6 +129,10 @@ static void *_thread(void *args)
 void dhcpv6_client_init(event_queue_t *eq, uint16_t netif)
 {
     assert(eq->waiter != NULL);
+#ifdef MODULE_GNRC_DHCPV6_CLIENT_MUD_URL
+    assert(strlen(mud_url) <= MAX_MUD_URL_LENGTH);
+    assert(strncmp(mud_url, "https://", 8) == 0);
+#endif  /* MODULE_GNRC_DHCPV6_CLIENT_MUD_URL */
```

Comment:
Shouldn't

```suggestion
    if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_MUD_URL)) {
        assert(strlen(mud_url) <= MAX_MUD_URL_LENGTH);
        assert(strncmp(mud_url, "https://", 8) == 0);
    }
```

work here as well?